aws/aws_kms/aws_kms.go

```diff
@@ -84,22 +85,34 @@ func New(
 	if err != nil {
 		return nil, fmt.Errorf("Failed to get credentials: %v", err)
 	}
-	config := &aws.Config{
-		Credentials: creds,
-		Region:      &region,
+	credProv := credentialsToProvider(creds)
+	config := aws.Config{
+		Credentials: credProv,
+		Region:      region,
 	}
-	sess := session.New(config)
-	kmsClient := kms.New(sess)
+
+	kmsClient := kms.NewFromConfig(config)
+
 	return &awsKmsSecrets{
 		client: kmsClient,
-		sess:   sess,
 		creds:  creds,
 		cmk:    cmk,
 		asc:    asc,
 		ps:     ps,
 	}, nil
 }
 
+func credentialsToProvider(creds *aws.Credentials) aws.CredentialsProvider {
+	return credentials.StaticCredentialsProvider{
```

Comment:
Lets not use a static credentials provider by default here. If say an ec2 role instance provider was used while generating the `aws.Credentials` object, then we should use that CredentialsProvider. In the file aws/credentials/credentials.go, introduce a new function `GetCredentialsProvider()` and return the provider that was created as part of `NewAWSCredentials` invocation.

Reply:
Addressed in this [commit](https://github.com/libopenstorage/secrets/pull/76/commits/6a1cf46cb0faa8455010bce92100ac7ce4d99d69)